skiplang/docs/SKStore.md

```diff
@@ -0,0 +1,142 @@
+# SKStore: A generic framework for reactive programming
+
+SKStore is a store made of collections called `directories`.
+Each directory contains entries associating values to a key.
+The keys can be of any type as long as they extend the class `SKStore.Key`, the values can be of any type as long as they extend the class `SKStore.File`.
+The APIs available for `SKStore` can be found in `Context.sk` and `Handle.sk`.
+
+## DirName
+
+Each directory (or collection) has a name, the name is defined with a slash separated path "/of/this/form/".
+The way to create a `DirName` is by using the primitive `DirName::create`.
+A directory name must start and finish with a slash, if it doesn't, an exception is thrown.
+The names are not used directly when manipulating directories, however, they come in handy when trying to debug the state that the system is in.
+One can navigate all the reactive values using familiar primitives (such as cd, ls etc ...).
+
+## Input directories
+
+Input directories are directories that can be modified by the outside world.
+To create one, you must use the primitive `mkdir`, the type of the key and of the values must be specified:
+
+```
+myInputDirectory =
+  context.mkdir(Type1::keyType, Type2::type, "/myInput/Name/")
+```
+
+Note that the fields "keyType" and "type" are automatically defined when one extends `SKStore.Key` and `SKStore.File` respectively.
+
+Once a directory is created, one can modify it (add/remove):
+
+```
+myInputDirectory.writeArray(context, myKey, Array[myValue])
+```
+
+Removing an element just consists of writing an empty array (also called a "tomb").
+There is no semantic difference between an empty array and no entry, since SKStore returns an empty array when trying to access a missing entry.
+
+Tombs are useful to handle concurrency and to synchronize directories with the outside world.
+They are collected automatically by the system.
```

Comment:
Is this relevant here? Feels more of an implementation detail to me

Reply:
At this point, this whole file is documenting the implementation, so seems ok.